channels/channels.go

```diff
@@ -0,0 +1,87 @@
+package channels
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/jaimeteb/chatto/channels/messages"
+	"github.com/jaimeteb/chatto/channels/rest"
+	"github.com/jaimeteb/chatto/channels/slack"
+	"github.com/jaimeteb/chatto/channels/telegram"
+	"github.com/jaimeteb/chatto/channels/twilio"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+// Config struct combines all available client configurations
+type Config struct {
+	Telegram telegram.Config `mapstructure:"telegram"`
+	Twilio   twilio.Config   `mapstructure:"twilio"`
+	Slack    slack.Config    `mapstructure:"slack"`
+}
+
+// Channels combines all available channel clients
+type Channels struct {
+	Telegram Channel
+	Twilio   Channel
+	REST     Channel
+	Slack    Channel
+}
+
+// Channel interface implements a channel to send and receive messages on
+type Channel interface {
+	// ReceiveMessage from the channel
+	ReceiveMessage(w http.ResponseWriter, r *http.Request) (*messages.Receive, error)
+	// ReceiveMessages from the channel. Starts a long running process, receives questions and sends them to the receiveChan
+	ReceiveMessages(receiveChan chan messages.Receive)
```

Comment:
Added a new function `ReceiveMessages` which is meant to be a long running process. I think the name is ok but fine with changing it. It takes a channel which it will send messages to the bot on. The bot will process the messages.

Reply:
Thank you! This is ideal for socket connections!